src/psyclone/psyGen.py

```diff
@@ -6193,6 +6477,53 @@ def __init__(self, name, parent=None):
         self._name = name
         self._symbol_table = SymbolTable(self)
 
+    @staticmethod
+    def create(name, symbol_table, children):
+        '''Create a Container instance given a name, a symbol table and a
+        list of child nodes.
+
+        :param str name: the name of the Container.
+        :param symbol_table: the symbol table associated with this \
+            Container.
+        :type symbol_table: :py:class:`psyclone.psyGen.SymbolTable`
+        :param children: a list of PSyIR nodes contained in the \
+            Container. These must be Containers or KernelSchedules.
+        :type children: list of :py:class:`psyclone.psyGen.Container` \
+            or :py:class:`psyclone.psyGen.KernelSchedule`
+
+        :returns: a Container instance.
+        :rtype: :py:class:`psyclone.psyGen.Container`
+
+        '''
+        if not isinstance(name, str):
+            raise GenerationError(
+                "name argument to class Container method create "
+                "should be a string but found '{0}'."
+                "".format(type(name).__name__))
+        if not isinstance(symbol_table, SymbolTable):
+            raise GenerationError(
+                "symbol_table argument to class Container method create "
+                "should be a SymbolTable but found '{0}'."
+                "".format(type(symbol_table).__name__))
+        if not isinstance(children, list):
+            raise GenerationError(
+                "children argument to class Container method "
+                "create should be a list but found '{0}'."
+                "".format(type(children).__name__))
+        for child in children:
+            if not isinstance(child, (KernelSchedule, Container)):
+                raise GenerationError(
+                    "child of children argument to class Container method "
+                    "create should be a PSyIR KernelSchedule or Container "
+                    "but found '{0}'.".format(type(child).__name__))
+        container = Container(name)
+        container._symbol_table = symbol_table
+        symbol_table._schedule = container
+        for child in children:
+            child.parent = container
+        container.children = children
+        return container
+
```

Comment:
Aesthetic comment, feel free to ignore, but I think a line-break between the input checking conditions and the body of the `create` method will be good to add into all the create methods

Reply:
Your wish is my command.